apps/studio/src/common/appdb/models/saved_connection.ts

```diff
@@ -154,6 +158,8 @@ export class DbConnectionBase extends ApplicationEntity {
       return '/var/run/mysqld/mysqld.sock'
     } else if (this.connectionType === 'postgresql') {
       return '/var/run/postgresql'
+    } else if (this.connectionType === 'tidb') {
+      return '/tmp/tidb.sock'
```

Comment:
From the TiDB code:

https://github.com/pingcap/tidb/blob/222fe17e2537d47498d7f5b6b79f5086b978eb80/cmd/tidb-server/main.go#L211

https://github.com/pingcap/tidb/blob/222fe17e2537d47498d7f5b6b79f5086b978eb80/pkg/server/server.go#L320-L323

From the `./bin/tidb-server` output:
```
[2024/03/15 13:37:00.424 +01:00] [INFO] [server.go:309] ["server is running MySQL protocol"] [addr=0.0.0.0:4000]
[2024/03/15 13:37:00.424 +01:00] [INFO] [server.go:323] ["server is running MySQL protocol"] [socket=/tmp/tidb-4000.sock]
```

Using the socket:
```
$ mysql -S /tmp/tidb-4000.sock -u root -BNe 'SELECT VERSION()'
8.0.11-TiDB-v8.0.0-alpha-535-ga6b1dd38ac
```

From the documentation:
https://docs.pingcap.com/tidb/stable/command-line-flags-for-tidb-configuration#--socket-string

There is also a system variable for this:
```
$ mysql -S /tmp/tidb-4000.sock -u root -e 'SHOW GLOBAL VARIABLES LIKE "socket"'
+---------------+---------------------+
| Variable_name | Value               |
+---------------+---------------------+
| socket        | /tmp/tidb-4000.sock |
+---------------+---------------------+
```

The docs for the variable: https://docs.pingcap.com/tidb/stable/system-variables#socket

This was added in TiDB v3.0 (released in 2019): https://docs.pingcap.com/tidb/stable/release-3.0-ga#tidb

Note that TiDB also supports `auth_socket`: https://docs.pingcap.com/tidb/stable/security-compatibility-with-mysql#authentication-plugin-status

Note that the default used to be `/tmp/tidb.sock`, but has changed to `/tmp/tidb-{Port}.sock`, which with the default `Port` of 4000 becomes `/tmp/tidb-4000.sock`. 

As TiDB is a distributed database it isn't common to use a UNIX socket to connect. However this is useful for CI setups and for securely bootstrapping etc.

Reply:
Note that MySQL also uses `/tmp` by default (see https://dev.mysql.com/doc/refman/8.0/en/connection-options.html#option_general_socket ). This might be different depending on the OS specific packages and whether they are from the OS or from the MySQL repositories.

And yes, `/tmp` isn't a very good location nowadays. Looks like `/run/` is a better place (see https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch03s15.html ). However this is all of historical reason from before `/run` was a thing.